validator/docs/schemas.md

```diff
@@ -0,0 +1,163 @@
+# Configuration Validator
+
+This module contains the logic needed to validate deployment files with a defined schema by Nestor.
+
+The deployment files define applications or projects.
+
+## Application Configuration
+
+In each file you will define the application that you want to deploy, there are few fields which are required to setup a new application.
+
+This is a basic configuration application with the minimum required fields.
+
+```yaml
+app: my-application   # Application's name
+is_enabled: false     # Is application deployment enabled
+git: git@microservice_repo # Git repository URL
+variables:            # Environment variables
+  app:
+    APP_VAR1: application_variable_1
+  ope:
+    OPE_VAR1: ope_variable_1
+```
+
+To customize your application deployment, there are other sections supported. Check the following sections.
+
+### Processes
+
+Define processes of a project. A project can contain multiple processes such as workers, web processes and more.
+The structure to define a process is the following:
+
+```yaml
+processes:
+  - name: process-name  # Process name, STRING - Required
+    is_cronjob: false   # Defining a simple process, BOOLEAN - Required
+    start_command: npm start  # Executed command when the pod is started. STRING - Required
+  - name: cronjob-name  # Process name - Required
+    is_cronjob: true    # Defining a scheduled process (cronjob) - Required
+    start_command: npm run worker # Executed command when the pod is started - Required
+```
+
+### Scales
+When you deploy an application in a Kubernetes environment, you define how many replicas you would like to run. 
+Each replica represents a Kubernetes POD that encapsulates your application container. 
+And also you can increase, or reduce the number of replicas in the given limit.
+
+```yaml
+scales:
+  my-process:         # Process to scale
+    min-replicas: 1   # Minimal number of pods, INTEGER - Required
+    max-replicas: 10  # Minimal number of pos, INTEGER - Required
+    targetCPUUtilizationPercentage: # Percentage of CPU which will trigger a horizontal scaling when it is exceeded, INTEGER, 0 <= x <= 100 - Required
```

Comment:
Is it possible to rename it or is this a fixed name ?
```suggestion
    targetCpuUtilizationPercentage: # Percentage of CPU which will trigger a horizontal scaling when it is exceeded, INTEGER, 0 <= x <= 100 - Required
```

Reply:
Also, add a value before the comment to make the block a valid yaml (e.g. 95)

Reply:
I think it would be a good time if we are going to keep the naming for acronyms in variables for example: HTML vs Html, CPU vs Cpu, and so on. In my opinion I like better to keep the acronyms in uppercase as long as they are not at the beginning of the variable definition.